### PR TITLE
nix develop shellHook for yarn install and starship

### PR DESCRIPTION
**Pull request by Kodylow**
adds yarn install check when entering `nix develop` and starship to shell, turbo bin gets installed by yarn install instead of nix.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -22,10 +21,15 @@
...
               pkgs.nodejs
               pkgs.yarn
+              pkgs.starship
               fedimint.packages.${system}.devimint
               fedimint.packages.${system}.gateway-pkgs
               fedimint.packages.${system}.fedimint-pkgs
             ] ++ prev.nativeBuildInputs;
+            shellHook = ''
+              yarn install
+              eval "$(starship init bash)"
```
[the remark this answers is not in the document]
Comment (Kodylow) [reply flake.nix:31]: Starship will overwrite it, yarn install will just run the command, can remove starship as it's a personal preference.

[the remark this answers is not in the document]
Comment (justinmoon) [reply flake.nix]: IMO we shouldn't be installing custom shells in our dev environments. As much as possible `nix develop` should feel like it's their normal shell IMO.

[the remark this answers is not in the document]
Comment (Kodylow) [reply flake.nix]: agree, removed in next commit.

**Review by wbobeirne — APPROVED** [review]

**Comment by wbobeirne** [comment]
Hope you don't mind the squash, last commit didn't match the semantic commits and it was effectively just a revert.

**Pull request merged**